signalfx/resource_signalfx_team_links.go

```diff
@@ -0,0 +1,183 @@
+package signalfx
+
+import (
+	"context"
+	"log"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func teamLinksResource() *schema.Resource {
+	return &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"team": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Team ID to link the configured dashboard groups or detectos to.",
+			},
+			"dashboard_groups": {
+				Type:        schema.TypeSet,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Description: "Dashboard group IDs to link to the team.",
+				Optional:    true,
+			},
+			"detectors": {
+				Type:        schema.TypeSet,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Description: "Detector IDs to link to the team.",
+				Optional:    true,
+			},
+		},
+		Create: teamLinksCreate,
+		Read:   teamLinksRead,
+		Update: teamLinksUpdate,
+		Delete: teamLinksDelete,
+		Exists: teamLinksExists,
+	}
+}
+
+func teamLinksExists(d *schema.ResourceData, meta interface{}) (bool, error) {
+	return true, nil
+}
+
+func teamLinksRead(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*signalfxConfig)
+	teamId := d.Get("team").(string)
+	res, err := config.Client.GetTeam(context.TODO(), teamId)
+	if err != nil {
+		if strings.Contains(err.Error(), "404") {
+			d.SetId("")
+		}
+		return err
+	}
+
+	_, new := d.GetChange("detectors")
+	detectors := new.(*schema.Set)
+
+	dets := make([]interface{}, len(res.Detectors))
+	for i, detector := range res.Detectors {
+		dets[i] = detector
+	}
+
+	teamDets := schema.NewSet(schema.HashString, dets)
+	// XXX: For some reason the same sets are hashing to different values (maybe to interface{} vs string types).
+	// Rehash the set so intersection, etc. will work.
+	detectors = schema.NewSet(schema.HashString, detectors.List())
+
+	if err := d.Set("detectors", teamDets.Intersection(detectors)); err != nil {
```

Comment:
The logic leading up to here is not very easy to follow.  Maybe rename some stuff or move some stuff out to helper functions.  It really isn't obvious what two things are being intersected here.